src/main/java/org/prebid/server/bidder/alkimi/AlkimiBidder.java

```diff
@@ -0,0 +1,176 @@
+        List<Imp> imps = request.getImp();
+        final List<Imp> updatedImps = new ArrayList<>(imps.size());
+
+        imps.forEach(imp -> {
+            ExtImpAlkimi ext = parseImpExt(imp);
+            updatedImps.add(updateImp(imp, ext));
+        });
+
+        final BidRequest outgoingRequest = request.toBuilder().imp(updatedImps).build();
+        return Result.of(Collections.singletonList(
+                        HttpRequest.<BidRequest>builder()
+                                .method(HttpMethod.POST)
+                                .uri(endpointUrl)
+                                .headers(HttpUtil.headers())
+                                .payload(outgoingRequest)
+                                .body(mapper.encodeToBytes(outgoingRequest))
+                                .build()
+                ),
+                Collections.emptyList()
+        );
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse), Collections.emptyList());
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                .collect(Collectors.toList());
+    }
+
+    private static BidType getBidType(String impId, List<Imp> imps) {
+        BidType bidType = BidType.banner;
+        for (Imp imp : imps) {
+            if (imp.getId().equals(impId)) {
+                if (imp.getBanner() != null) {
+                    return bidType;
+                } else if (imp.getVideo() != null) {
+                    bidType = BidType.video;
+                } else if (imp.getXNative() != null) {
+                    bidType = BidType.xNative;
+                } else if (imp.getAudio() != null) {
+                    bidType = BidType.audio;
+                }
+            }
+        }
+        return bidType;
+    }
+
+    private ExtImpAlkimi parseImpExt(Imp imp) {
```

Comment:
Please, place method definitions in call order(i.e. for this method: `place it after makeHttpRequests`), also, please, check for similar occurences.